internal/services/reservations_service.go

```diff
@@ -44,57 +45,64 @@ func CreateReservation(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// validate pubkey
-	if payload.Pubkey.ExistingID == nil &&
-		payload.Pubkey.NewName == nil &&
-		payload.Pubkey.NewBody == nil {
-		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyNewError))
-	}
-	if payload.Pubkey.ExistingID != nil &&
-		payload.Pubkey.NewName != nil &&
-		payload.Pubkey.NewBody != nil {
-		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyNewError))
-	}
-	if payload.Pubkey.ExistingID == nil && (payload.Pubkey.NewName == nil || payload.Pubkey.NewBody == nil) {
-		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyMissingError))
+	if payload.Provider != models.ProviderTypeNoop {
+		if payload.Pubkey.ExistingID == nil &&
+			payload.Pubkey.NewName == nil &&
+			payload.Pubkey.NewBody == nil {
+			renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyNewError))
+		}
+		if payload.Pubkey.ExistingID != nil &&
+			payload.Pubkey.NewName != nil &&
+			payload.Pubkey.NewBody != nil {
+			renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyNewError))
+		}
+		if payload.Pubkey.ExistingID == nil && (payload.Pubkey.NewName == nil || payload.Pubkey.NewBody == nil) {
+			renderError(w, r, payloads.NewInvalidRequestError(r.Context(), InvalidRequestPubkeyMissingError))
+		}
 	}
 
 	reservation := &models.Reservation{
+		Provider:  payload.Provider,
 		AccountID: accountId,
 		Status:    "Created",
 	}
 
 	// create or validate pubkey
 	var pk *models.Pubkey
-	if payload.Pubkey.ExistingID == nil {
-		pk = &models.Pubkey{
-			AccountID: accountId,
-			Name:      *payload.Pubkey.NewName,
-			Body:      *payload.Pubkey.NewBody,
-		}
-		err = pkDao.Create(r.Context(), pk)
-		if err != nil {
-			renderError(w, r, payloads.NewDAOError(r.Context(), "create pubkey", err))
-			return
-		}
-		logger.Debug().Msgf("Created a new pubkey %d named '%s'", pk.ID, pk.Name)
-	} else {
-		// TODO: IMPORTANT! Must perform GetByIdAndAccount to prevent pubkey hijack
-		logger.Debug().Msgf("Validating existing pubkey %d", *payload.Pubkey.ExistingID)
-		pk, err = pkDao.GetById(r.Context(), *payload.Pubkey.ExistingID)
-		if err != nil {
-			var e *dao.NoRowsError
-			if errors.As(err, &e) {
-				renderError(w, r, payloads.NewNotFoundError(r.Context(), err))
-			} else {
-				renderError(w, r, payloads.NewDAOError(r.Context(), "get pubkey by id", err))
+	if payload.Provider != models.ProviderTypeNoop {
```

Comment:
I get it would be bit overkill to add another table for noop, but given the service code is quite different anyway, wouldn't it be better to have special enpoint for noop?